knack/introspection.py

```diff
@@ -67,45 +67,23 @@ def extract_args_from_signature(operation, excluded_params=None):
     """ Extracts basic argument data from an operation's signature and docstring
         excluded_params: List of params to ignore and not extract. By default we ignore ['self', 'kwargs'].
     """
-    args = []
-    try:
-        # only supported in python3 - falling back to argspec if not available
-        sig = inspect.signature(operation)
-        args = sig.parameters
-    except AttributeError:
-        sig = inspect.getargspec(operation)  # pylint: disable=deprecated-method, useless-suppression
-        args = sig.args
+    sig = inspect.signature(operation)
+    args = sig.parameters
```

Comment:
[`inspect.getargspec`](https://docs.python.org/3.10/library/inspect.html#inspect.getargspec) is removed in Python 3.11 and replaced by [`inspect.getfullargspec`](https://docs.python.org/3.11/library/inspect.html#inspect.getfullargspec). However, `inspect.getfullargspec` is retained only to maintain backward compatibility. The recommendation is to use [`inspect.signature`](https://docs.python.org/3.11/library/inspect.html#inspect.signature):

> Note that [signature()](https://docs.python.org/3.11/library/inspect.html#inspect.signature) and [Signature Object](https://docs.python.org/3.11/library/inspect.html#inspect-signature-object) provide the recommended API for callable introspection, and support additional behaviours (like positional-only arguments) that are sometimes encountered in extension module APIs. This function is retained primarily for use in code that needs to maintain compatibility with the Python 2 inspect module API.
> -- https://docs.python.org/3.11/library/inspect.html#inspect.getfullargspec

Given we have dropped Python 2 long ago (https://github.com/microsoft/knack/pull/233), it's time to drop the usage of `inspect.getargspec` too.